common/src/stack/probepal/bin/probepal.py

```diff
@@ -0,0 +1,45 @@
+#! /opt/stack/bin/python3
+
+import sys
+import json
+from operator import attrgetter
+import dataclasses
+
+import stack.probepal
+
+# make dataclasses serializable
+class EnhancedJSONEncoder(json.JSONEncoder):
+        def default(self, o):
+                if dataclasses.is_dataclass(o):
+                        return dataclasses.asdict(o)
+                return super().default(o)
+
+# general:
```

Comment:
Can packages/modules have docstrings? If so I think this would be a good candidate for one.